examples/DataApi-Water-And-Energy-Usage/template/modules/shared_data/src/main/scala/com/my/currency/shared_data/Main.scala

```diff
@@ -12,69 +12,218 @@ import org.http4s.HttpRoutes
 import org.tessellation.currency.dataApplication.dataApplication.DataApplicationValidationErrorOr
 import org.tessellation.currency.dataApplication.{DataApplicationValidationError, DataState, DataUpdate}
 import org.tessellation.security.signature.Signed
+import cats.syntax.all._
+import com.my.currency.shared_data.Data.{AggregatedUsage, EnergyUsage, WaterUsage}
+import com.my.currency.shared_data.Validations.{validateEnergyTimestampToNoDoubling, validateEnergyUsageUpdate, validateProvidedAddress, validateWaterTimestampToNoDoubling, validateWaterUsageUpdate}
+import monocle.syntax.all._
+import org.tessellation.schema.address.Address
+import org.tessellation.security.SecurityProvider
 
 import java.nio.charset.StandardCharsets
 
+object Errors {
+  case object EnergyNotPositive extends DataApplicationValidationError {
+    val message = "energy usage must be positive"
+  }
+
+  case object EnergyAlreadyUpdated extends DataApplicationValidationError {
+    val message = "Energy already updated, skipping"
```

Comment:
What's actually happening in this case is that the Energy update is older than the most recent update so it's rejected. How about `EnergyUpdateOutdated` with message "Energy update older than latest update timestamp, rejecting" 

+ similar pattern for Water